plugins/lookup/nb_lookup.py

```diff
@@ -222,11 +254,13 @@ def run(self, terms, variables=None, **kwargs):
 
         results = []
         for term in terms:
-
-            try:
-                endpoint = get_endpoint(netbox, term)
-            except KeyError:
-                raise AnsibleError("Unrecognised term %s. Check documentation" % term)
+            if netbox_plugin:
```

Comment:
I believe we should be doing a try/except block here and then return an error if we hit an `AttributeError`.

Return something like: `AnsibleError("%s does not exist in %s plugin." % (term, plugin)`

This is because `get_plugin_endpoint` uses the `getattr` which can return an `AttributeError`

Reply:
Checking for an `AttributeError` doesn't work here, unfortunately. Because these plugin models aren't build into the core of pynetbox, the endpoint seems to be merely a string used to craft the api url and only eval'd at request time. Any plugin/model name appears to be valid, regardless of whether the netbox instance actually has it.

There is an `installed_plugins()` call that can be made to at least check for a valid plugin name but I don't see an option for verifying the model itself:
```
[{'name': 'Test', 'package': 'netbox_test', 'author': 'test', 'author_email': 'test@test.com', 'description': 'test plugin', 'verison': '0.1'}]

Reply:
Ah yes. Thanks for the explanation of that.

I'm thinking we may want to wrap the `filter` and `all` in a try/except block looking for a `RequestError` and then maybe inspect a bit further for a 404 error and then provide a meaningful error message such as; `Plugin endpoint is not valid. Please validate the provided plugin has the requested endpoint. {url}` or something like that.